objectivec/GPBWellKnownTypes.m

```diff
@@ -115,3 +115,40 @@ - (void)setTimeIntervalSince1970:(NSTimeInterval)timeIntervalSince1970 {
 }
 
 @end
+
+NSString *const GPBTypeGoogleApisComPrefix = @"type.googleapis.com/";
+
+@implementation GPBAny (GBPWellKnownTypes)
+
+- (NSString*)getAnyTypeName {
+  NSAssert([self.typeURL hasPrefix:GPBTypeGoogleApisComPrefix], @"Invalid any type url.");
+  return [self.typeURL substringFromIndex: [GPBTypeGoogleApisComPrefix length]];
+}
+
+- (void)unpackTo:(GPBMessage*)message {
+  NSAssert([self is:[message class]],
+           [NSString stringWithFormat:@"%@/%@/%@/%@/%@",
```

Comment:
Why not just

[NSString stringWithFormat:@"Given messsage's type name (@%) doesn't match with any's (%@)", message.descriptor.name, self.getAnyTypeName)